src/quo2/components/tags/tags.cljs

```diff
@@ -1,29 +1,142 @@
+                                                            (when on-scroll
+                                                              (on-scroll e)))
+                       :render-fn                         (fn [{:keys [id label resource]} index]
+                                                            [rn/view {:style {:margin-right  (if (= size default-tab-size) 12 8)
+                                                                              :padding-right (when (= index (dec (count data)))
+                                                                                               (get-in props [:style :padding-left]))}}
+                                                             [tag/tag {:id            id
+                                                                       :size          size
+                                                                       :active        (= id @active-tab-id)
+                                                                       :resource      (if (= type :emoji)
+                                                                                        resource
+                                                                                        :main-icons2/placeholder)
+                                                                       :blurred?      blurred?
+                                                                       :icon-color    icon-color
+                                                                       :disabled?     disabled?
+                                                                       :label         label
+                                                                       :type          type
+                                                                       :labelled?     (if (= :label type) true labelled?)
+                                                                       :on-press      (fn [id]
+                                                                                        (reset! active-tab-id id)
+                                                                                        (when scroll-on-press?
+                                                                                          (.scrollToIndex @flat-list-ref
+                                                                                                          #js {:animated     true
+                                                                                                               :index        index
+                                                                                                               :viewPosition 0.5}))
+                                                                                        (when on-change
+                                                                                          (on-change id)))}
+                                                              label]])})])))))
```

Comment:
Can't we just do
```clj
[tabs/scrollable-tabs {:scroll-on-press?    true
                       :fade-end?           true
                       :default-active      1
                       :data                [{:id 1 :label "Music"     :resource (resources/get-image :music)}
                                             {:id 2 :label "Lifestyle" :resource (resources/get-image :lifestyle)}
                                             {:id 2 :label "Podcasts"  :resource (resources/get-image :podcasts)}
                                             {:id 2 :label "Music"     :resource (resources/get-image :music)}
                                             {:id 3 :label "Lifestyle" :resource (resources/get-image :lifestyle)}]
                       :size                32
                       :labelled?           true
                       :type                :emoji
                       :fade-end-percentage 0.4}]
                       ```

Reply:
I mean, If it's the same logic .. We must reuse it.

Reply:
@ibrkhalil  this would mean refactoring tab and tag component and we might end up with a very complex component which is something we are also trying to avoid with the redesign. I think its better to have each tag component in separate ns. However what we can is place scrollable-tabs/tags into a separate ns so that both tags and tabs can share the same logic.

Reply:
Got it, thanks John
If you needed any backup .. I'm happy to help